R/ard_hierarchical.R

```diff
@@ -93,6 +93,10 @@ ard_hierarchical.data.frame <- function(data,
     return(dplyr::tibble() |> as_card())
   }
 
+  # check factor levels --------------------------------------------------------
+  check_no_na_factor_levels(data[c(variables, by)])
+  check_factor_has_levels(data[c(variables, by)])
+
```

Comment:
It looks like we can remove this check, because the `variables` and `by` arguments are passed to `ard_categorical()` a few lines later.

Reply:
Factor checks removed from `ard_hierarchical()`